src/App.jsx

```diff
@@ -1,105 +1,70 @@
-import React, { useState, useEffect, useCallback } from 'react'
+import React, { useCallback, useEffect, useState } from 'react'
 
-import './App.css'
-import 'bootstrap/dist/css/bootstrap.css'
 import '@fortawesome/fontawesome-free/css/all.css'
+import 'bootstrap/dist/css/bootstrap.css'
+import './App.css'
 
-import Login from './components/Login'
 import JsPsychExperiment from './components/JsPsychExperiment'
+import Login from './components/Login'
 
-import { getProlificId, sleep } from './lib/utils'
-import { initParticipant, addToFirebase } from './firebase'
 import { config, taskVersion, turkUniqueId } from './config/main'
+import { addToFirebase, validateParticipant } from './firebase'
+import { getProlificId, sleep } from './lib/utils'
 
+/**
+ * The top-level React component for Honeycomb. App handles initiating the jsPsych component when the participant
+ * successfully logs in, and stores the overall state of the experiment. Importantly, App keeps track of what the
+ * experiment is running on (Electron, Firebase, PsiTurk, or MTurk).
+ *
+ * Note that App is a functional component, which means it uses React callbacks rather than class methods. Learn more
+ * about functional vs. class components here: https://reactjs.org/docs/components-and-props.html. It is recommended
+ * to use functional components.
+ */
 function App () {
+  // Manage if a user is currently logged in
+  const [loggedIn, setLoggedIn] = useState(false)
+  // Manage error state of the app
+  const [isError, setIsError] = useState(false)
+
+  // Manage the electron renderer
+  const [ipcRenderer, setIpcRenderer] = useState()
+  // Manage the psiturk object
+  const [psiturk, setPsiturk] = useState(false)
+
+  // Manage the data used in the experiment
+  const [participantID, setParticipantID] = useState('')
+  const [studyID, setStudyID] = useState('')
   // Remember startDate between renders.
   // This is necessary to allow Firebase to create a timestamped document at Login time,
   // and then to find and update the *same* timestamped document after each trial in JsPsychExperiment.
-  const [startDate] = useState(new Date().toISOString())
+  // TODO: Shouldn't the document have a UID? Save the date but we should have a uid for the document?
```

Comment:
Doesn't the UID get automatically generated? Or am I wrong about that?

Reply:
I think in honeycomb we use the startDate as a document identifier. But I also think that from their [best practices](https://firebase.google.com/docs/firestore/best-practices), I don't know if dates qualify as monotonically increasing and therefore if that's a good idea or not. I think we did it that way for user readability and possibly to be consistent with that is done on desktop. But be could revisit

Reply:
Looks like firestore will create an ID if you don't feed it one
![image](https://user-images.githubusercontent.com/38439940/223491366-36895f55-f83d-413d-bd0e-7796ad90c9dc.png)